web/src/components/volume.tsx

```diff
@@ -1,29 +1,63 @@
 import { createSignal } from "solid-js"
 
-type VolumeButtonProps = {
+type VolumeControlProps = {
 	mute: (isMuted: boolean) => void
+	setVolume: (newVolume: number) => void
 }
 
-export const VolumeButton = (props: VolumeButtonProps) => {
+export const VolumeControl = (props: VolumeControlProps) => {
 	const [isMuted, setIsMuted] = createSignal(false)
+	const [currentVolume, setCurrentVolume] = createSignal(1)
 
 	const toggleMute = () => {
 		const newIsMuted = !isMuted()
 		setIsMuted(newIsMuted)
-		props?.mute(newIsMuted)
+		props.mute(newIsMuted)
+
+		if (newIsMuted) {
+			props.setVolume(0)
+			setCurrentVolume(0)
+		} else {
+			props.setVolume(1)
+			setCurrentVolume(1)
```

Comment:
@JoaquinBCh After unmute the volume level should be set to the previous level before mutting.